src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -823,6 +823,56 @@ def deviceAvailable(self, device, err):
 
         return DeviceProxyWrapper(returnDevice, self._dmLib)
 
+    async def GetConnectedDevice(self, nodeid, allowPASE=True, timeoutMs: int = None):
+        ''' Returns DeviceProxyWrapper upon success.'''
+        self.CheckIsActive()
+
+        if allowPASE:
+            returnDevice = c_void_p(None)
+            res = self._ChipStack.Call(lambda: self._dmLib.pychip_GetDeviceBeingCommissioned(
+                self.devCtrl, nodeid, byref(returnDevice)), timeoutMs)
+            if res.is_success:
+                logging.info('Using PASE connection')
+                return DeviceProxyWrapper(returnDevice)
+
+        eventLoop = asyncio.get_running_loop()
+        future = eventLoop.create_future()
+
+        class DeviceAvailableClosure():
+            def __init__(self, loop, future: asyncio.Future):
+                self._returnDevice = c_void_p(None)
+                self._returnErr = None
+                self._event_loop = loop
+                self._future = future
+
+            def _deviceAvailable(self):
+                if self._returnDevice.value is not None:
+                    self._future.set_result(self._returnDevice)
+                else:
+                    self._future.set_exception(self._returnErr.to_exception())
+
+            def deviceAvailable(self, device, err):
+                self._returnDevice = c_void_p(device)
+                self._returnErr = err
+                self._event_loop.call_soon_threadsafe(self._deviceAvailable)
+                ctypes.pythonapi.Py_DecRef(ctypes.py_object(self))
+
+        closure = DeviceAvailableClosure(eventLoop, future)
+        ctypes.pythonapi.Py_IncRef(ctypes.py_object(closure))
+        self._ChipStack.Call(lambda: self._dmLib.pychip_GetConnectedDeviceByNodeId(
+            self.devCtrl, nodeid, ctypes.py_object(closure), _DeviceAvailableCallback),
+            timeoutMs).raise_on_error()
+
+        # The callback might have been received synchronously (during self._ChipStack.Call()).
+        # In that case the Future has already been set it will return immediately
+        if (timeoutMs):
```

Comment:
nit: remove brackets. Should we use `is not None` or are we saying 0 and None are equivalent of "no timeout" ? I would generally be more inclined to think 0 is "run and return immediately" (even though maybe that seems useless).

Reply:
> Should we use is not None or are we saying 0 and None are equivalent of "no timeout" ? 

Yes. I followed `GetConnectedDeviceSync` semantic here.

I am fine changing this semantic, but it should be done in `GetConnectedDeviceSync` too. From what I can all users either leave the default or pass a positive value (even the YAML tests). So changing the semantic should be fine, at least for all uses within the Matter SDK.

Reply:
FWIW, changing this to 
```
        if timeoutMs is not None:
```

Would lead to a Timeout exception in almost all cases when passing 0. It could actually useful to test if/when the statement just above actually kicks in (callback received synchronously during self._ChipStack.Call()).